meson.build

```diff
@@ -34,7 +34,8 @@ openssl_dep = dependency('openssl', version: '>=1.1.1', required: get_option('cp
 if openssl_dep.found()
   deps += openssl_dep
   args += '-DCPPHTTPLIB_OPENSSL_SUPPORT'
-  if host_machine.system() == 'darwin'
+  if get_option('cpp-httplib_apple_keychain') and host_machine.system() == 'darwin'
+    args += '-DCPPHTTPLIB_USE_CERTS_FROM_APPLE_KEYCHAIN'
```

Comment:
I would keep the Meson (and CMake) code as is, as these libraries are always available on macOS (right?) and linking to them is only an issue when using the header without any build system handling the compilation

Reply:
I think when user use build system it was more correct to implicitly enable this feature via build system. So when I use cmake, HTTPLIB_IS_USING_CERTS_FROM_APPLE_KEYCHAIN feature wiil be enable implicitly. 
But when using header-only without any build scripts library user can be explicitly specify that the feature is required.
Finnaly I think that build system should also provide this option anyway.

Reply:
So it is strange when linker gets additional flags without any semantic artifacts in the build script